L1Trigger/L1THGCal/plugins/backend/HGCalTriggerClusterInterpretationEM.cc

```diff
@@ -0,0 +1,56 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterInterpreterBase.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+
+class HGCalTriggerClusterInterpretationEM : public HGCalTriggerClusterInterpreterBase {
+public:
+  HGCalTriggerClusterInterpretationEM();
+  ~HGCalTriggerClusterInterpretationEM() override{};
+  void initialize(const edm::ParameterSet& conf) final;
+  void eventSetup(const edm::EventSetup& es) final;
+  void interpret(l1t::HGCalMulticlusterBxCollection& multiclusters) const final;
+
+private:
+  std::vector<double> layer_containment_corrs_;
+  std::vector<double> scale_corrections_coeff_;
+  std::vector<double> dr_bylayer_;
+
+  HGCalTriggerTools triggerTools_;
+};
+
+DEFINE_HGC_TPG_CLUSTER_INTERPRETER(HGCalTriggerClusterInterpretationEM, "HGCalTriggerClusterInterpretationEM");
+
+HGCalTriggerClusterInterpretationEM::HGCalTriggerClusterInterpretationEM() {}
+
+void HGCalTriggerClusterInterpretationEM::initialize(const edm::ParameterSet& conf) {
+  layer_containment_corrs_ = conf.getParameter<std::vector<double>>("layer_containment_corr");
+  scale_corrections_coeff_ = conf.getParameter<std::vector<double>>("scale_correction_coeff");
```

Comment:
It could be checked that `scale_corrections_coeff_` have a size of 2

Reply:
done/